### PR TITLE
feat(bazel/markdown-to-html): support headings in markdown to html

### PR DESCRIPTION
**Pull request by josephperrott**
Properly handle headings for custom styling

### CODE REVIEW

File: bazel/markdown_to_html/src/tranformations/heading.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import {RendererApi} from 'marked';
+
+import {getPageTitle} from '../utils';
+import {getHeaderId} from '../state';
+
+export const headingRender: RendererApi['heading'] = (text, level, raw) => {
+  if (level === 1) {
+    return `
+    <!-- Document Header -->
```
Comment (jelbourn) [inline bazel/markdown_to_html/src/tranformations/heading.ts:17]: Is the comment necessary? (admittedly small, but still extra bytes over the wire) 

If it is load bearing in some way, we should have a comment that calls that out 

(similar for other html comments)

File: bazel/markdown_to_html/src/tranformations/heading.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  // regex aren't perfect for that but this one should be "good enough"
+  const regex = /<a\s+(?:[^>]*?\s+)?href.*?>(.*?)<\/a>/gi;
+  const anchorLessText = text.replace(regex, '$1');
+
+  const link = getHeaderId(anchorLessText);
+  const label = anchorLessText.replaceAll(/`(.*?)`/g, '<code>$1</code>');
+
+  return `
+  <h${level} id="${link}">
+    <a href="#${link}" class="docs-anchor" tabindex="-1">${label}</a>
```
Comment (jelbourn) [inline bazel/markdown_to_html/src/tranformations/heading.ts:39]: I don't think the anchor should be a child of the header. From an a11y standpoint, you would want the anchor to immediately follow the header in navigation order.

**Review by jelbourn — APPROVED** [review]
LGTM aside from the anchor nested inside headers

File: bazel/markdown_to_html/src/tranformations/heading.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  // regex aren't perfect for that but this one should be "good enough"
+  const regex = /<a\s+(?:[^>]*?\s+)?href.*?>(.*?)<\/a>/gi;
+  const anchorLessText = text.replace(regex, '$1');
+
+  const link = getHeaderId(anchorLessText);
+  const label = anchorLessText.replaceAll(/`(.*?)`/g, '<code>$1</code>');
+
+  return `
+  <h${level} id="${link}">${label}</h${level}>
+  <a href="#${link}" class="docs-anchor" tabindex="-1"></a>
```
Comment (jelbourn) [inline bazel/markdown_to_html/src/tranformations/heading.ts:38]: Maybe a TODO, but this anchor would also need either an `aria-label` or visually hidden content like `"Link to ${label}"`

**Comment by josephperrott** [comment]
This PR was merged into the repository by commit 7264838eafd6e989a1410c273e20bd0d90af2e39.

**Pull request closed without merging**